cylc/flow/scheduler.py

```diff
@@ -1500,6 +1512,8 @@ async def main_loop(self):
             # Useful for debugging core scheduler issues:
             # self.pool.log_task_pool(logging.CRITICAL)
 
+            self.process_command_queue()
```

Comment:
I'm not sure about moving this before the reload as I think it could cause commands to be run out of order.

E.G. Consider this sequence of events

```bash
cylc reload x
cylc trigger x//y/z
```

I think before the trigger would happen *after* the reload but after the change it would happen *before* it?

Reply:
Why do we process the command queue twice in 1 iteration?

Reply:
Hmm, I don't recall if there is a good reason for that.

Reply:
There will be a reason for this, buried long ago in the evolution of the main loop. The order of events in the main loop is quite sensitive and can impact stability and untested functionality. I wouldn't spend time trying to work out why it's the way it is just now, we will re-jig this when we make Cylc commands asynchronous (which would allow the CLI to wait for command completion before exiting).